src/rpcq.lisp

```diff
@@ -212,10 +213,10 @@ We distinguish between the following options for any field type:
 
     ;; handle lists
     ((eq ':list (car field-type))
-     ;; Need not check if REQUIRED as NIL is still of type list
-     ;; We use 'simple-vector rather than 'list as this maps better to
-     ;; the JSON distinction betweel null and []
-     (values 'simple-vector (coerce default 'simple-vector)))
```

Comment:
I'm surprised this is the only place this needed to change. Everywhere else was already hitting vectors? Do we have a test in the test suite that de/serializes a variety of data types? The call/response test payloads are quite simple, and I don't think that they involve any sequences.